src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -10,143 +13,335 @@ module Dialogs
+          # trigger disabling the other subwidgets
+          handle("ID" => value)
         end
 
+        # @macro seeAbstractWidget
         def store
-          start_block = Yast::UI.QueryWidget(Id(:start_block), :Value)
-          end_block = Yast::UI.QueryWidget(Id(:end_block), :Value)
-          len = end_block - start_block + 1
-          bsize = @slot.region.block_size # where does this come from?
-          region = Y2Storage::Region.create(start_block, len, bsize)
-          @ptemplate.region = region
-        end
-
-        def max_size_term
-          term(
-            :LeftRadioButton,
-            Id(:max_size),
-            Opt(:notify),
-            # radio button text, %1 is replaced by size
-            Builtins.sformat(
-              _("Maximum Size (%1)"),
-              @max_size.to_human_string
-            )
-          )
+          w = current_widget
+          w.store
+          @ptemplate.region = w.region
+          @ptemplate.size_choice = value
         end
+      end
 
-        def manual_size_term
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_size),
-            Opt(:notify),
-            _("Custom Size"),
-            VBox(
-              Id(:manual_size_attachment),
-              MinWidth(
-                15,
-                InputField(Id(:size_input), Opt(:shrinkable), _("Size"))
-              )
-            )
-          )
+      # An invisible widget that knows a Region
+      class MaxSizeDummy < CWM::Empty
+        attr_reader :region
+
+        def initialize(region)
+          @region = region
         end
 
-        def manual_region_term
-          min_block = @disk.region.start
-          # FIXME: libyui widget overflow :-(
-          max_block = min_block + @disk.region.length - 1
-          start_block = @slot.region.start
-          end_block = start_block + @slot.region.length - 1
-
-          # radio button text
-          term(
-            :LeftRadioButtonWithAttachment,
-            Id(:manual_region),
-            Opt(:notify),
-            _("Custom Region"),
-            VBox(
-              Id(:manual_region_attachment),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:start_block),
-                  _("Start Block"),
-                  min_block, max_block, start_block
-                )
-              ),
-              MinWidth(
-                10,
-                IntField(
-                  Id(:end_block),
-                  _("End Block"),
-                  min_block, max_block, end_block
-                )
+        # @macro seeAbstractWidget
+        def store
+          # nothing to do, that's OK
+        end
+      end
+
+      # Enter a human readable size
+      class CustomSizeInput < CWM::InputField
+        # @return [Y2Storage::DiskSize]
+        attr_reader :min_size, :max_size
+
+        def initialize(ptemplate, regions)
```

Comment:
please specify types of parameters. And also what is ptemplate, as it is not so obvious.